modules/host/host.go

```diff
@@ -14,43 +14,197 @@ import (
 	"github.com/NebulousLabs/Sia/persist"
 	"github.com/NebulousLabs/Sia/types"
+
+	"github.com/NebulousLabs/bolt"
 )
 
+// TODO: Write a document on how to use the host, suggest that RAID0 be used to
+// protect the core files from drive failure, emphasize that the host is pretty
+// relient on having information that is not outdated.
+
 const (
-	defaultTotalStorage = 10e9         // 10 GB.
-	defaultMaxDuration  = 144 * 30 * 6 // 6 months.
+	// defaultMaxDuration defines the maximum number of blocks into the future
+	// that the host will accept for the duration of an incoming file contract
+	// obligation. 6 months is chosen because hosts are expected to be
+	// long-term entities, and because we want to have a set of hosts that
+	// support 6 month contracts when Sia leaves beta.
+	defaultMaxDuration = 144 * 30 * 6 // 6 months.
+
+	// maximumStorageFolders indicates the maximum number of storage folders
+	// that the host allows. Some operations, such as creating a new storage
+	// folder, take longer if there are more storage folders. Static RAM usage
+	// also increases as the number of storage folders increase. For this
+	// reason, a limit on the maximum number of storage folders has been set.
+	maximumStorageFolders = 100
+
+	// resubmissionTimeout defines the number of blocks that a host will wait
+	// before attempting to resubmit a transaction to the blockchain.
+	// Typically, this transaction will contain either a file contract, a file
+	// contract revision, or a storage proof.
+	resubmissionTimeout = 3
+
+	// Names of the various persistent files in the host.
+	dbFilename   = modules.HostDir + ".db"
+	logFile      = modules.HostDir + ".log"
+	settingsFile = modules.HostDir + ".json"
 )
 
 var (
-	// defaultPrice defines the starting price for hosts selling storage. We
-	// try to match a number that is both reasonably profitable and reasonably
-	// competitive.
-	defaultPrice = modules.StoragePriceToConsensus(100e3) // 100 SC / GB / Month
+	// dbMetadata is a header that gets put into the database to identify a
+	// version and indicate that the database holds host information.
+	dbMetadata = persist.Metadata{
+		Header:  "Sia Host DB",
+		Version: "0.5.2",
+	}
+
+	// persistMetadata is the header that gets written to the persist file, and is
+	// used to recognize other persist files.
+	persistMetadata = persist.Metadata{
+		Header:  "Sia Host",
+		Version: "0.5",
+	}
+
+	// defaultContractPrice defines the default price of creating a contract
+	// with the host. The default is set to 50 siacoins, which means that the
+	// opening file contract can have 5 siacoins put towards it, the file
+	// contract revision can have 15 siacoins put towards it, and the storage
+	// proof can have 15 siacoins put towards it, with 5 left over for the
+	// host, a sort of compensation for keeping and backing up the storage
+	// obligation data.
+	defaultContractPrice = types.NewCurrency64(50).Mul(types.SiacoinPrecision) // 50 siacoins
```

Comment:
I don't understand what `ContractPrice` is used for. It doesn't appear to be used in this PR.